examples/random/src/main.cpp

```diff
@@ -20,115 +20,132 @@
-      auto x_acc = x_buffer.template get_access<sycl::access::mode::write>(cgh);
-
-      cgh.parallel_for(sycl::range<1>(x.size()), [=](sycl::item<1> idx) {
-        unsigned long long offset = idx.get_linear_id();
-
-        // Create minstd_rand engine
-        oneapi::dpl::minstd_rand engine(seed, offset);
-
-        // Create float uniform_real_distribution distribution with a = 0, b = 1
-        oneapi::dpl::uniform_real_distribution<float> distr;
-
-        // Generate float random number
-        auto res = distr(engine);
-
-        // Store results to x_acc
-        x_acc[idx] = res;
-      });
-    });
-  }
-
-  std::cout << "\nsuccess for scalar generation" << std::endl;
-  std::cout << "First 5 samples of minstd_rand with scalar generation"
-            << std::endl;
-  for (int i = 0; i < 5; i++) {
-    std::cout << x.begin()[i] << std::endl;
-  }
-
-  std::cout << "\nLast 5 samples of minstd_rand with scalar generation"
-            << std::endl;
-  for (int i = 0; i < 5; i++) {
-    std::cout << x.rbegin()[i] << std::endl;
-  }
-}
-
-template <int VecSize, typename RealType>
-void vector_example(sycl::queue& queue, std::vector<RealType>& x) {
-  {
-    sycl::buffer<RealType, 1> x_buffer(x.data(), sycl::range<1>(x.size()));
-
-    queue.submit([&](sycl::handler& cgh) {
-      auto x_acc = x_buffer.template get_access<sycl::access::mode::write>(cgh);
-
-      cgh.parallel_for<class generate_kernel>(
-          sycl::range<1>(x.size() / VecSize), [=](sycl::item<1> idx) {
-            unsigned long long offset = idx.get_linear_id() * VecSize;
-
-            // Create minstd_rand engine
-            oneapi::dpl::minstd_rand_vec<VecSize> engine(seed);
-            engine.discard(offset);
-
-            // Create float uniform_real_distribution distribution
-            oneapi::dpl::uniform_real_distribution<sycl::vec<float, VecSize>>
-                distr;
-
-            // Generate sycl::vec<float, VecSize> of random numbers
-            auto res = distr(engine);
-
-            // Store results from res to VecSize * offset position of x_acc
-            res.store(idx.get_linear_id(), x_acc.get_pointer());
-          });
-    });
-  }
-
-  std::cout << "\nsuccess for vector generation" << std::endl;
-  std::cout << "First 5 samples of minstd_rand with vector generation"
-            << std::endl;
-  for (int i = 0; i < 5; i++) {
-    std::cout << x.begin()[i] << std::endl;
-  }
-
-  std::cout << "\nLast 5 samples of minstd_rand with vector generation"
-            << std::endl;
-  for (int i = 0; i < 5; i++) {
-    std::cout << x.rbegin()[i] << std::endl;
-  }
-}
-
-int main() {
-  auto async_handler = [](sycl::exception_list ex_list) {
-    for (auto& ex : ex_list) {
-      try {
-        std::rethrow_exception(ex);
-      } catch (sycl::exception& ex) {
-        std::cerr << ex.what() << std::endl;
-        std::exit(1);
-      }
+void
+scalar_example(sycl::queue& queue, const ::std::uint32_t seed, std::vector<RealType>& x)
```

Comment:
we have so many changes with formatting this stuff. Do we really want to change formatting? Does it help?

Reply:
Yes, it's help for me and looks good for Pavel too.
You may see changes in commit diffs.